xbmc/addons/Service.cpp

```diff
@@ -20,89 +20,148 @@
 #include "Service.h"
 #include "AddonManager.h"
 #include "interfaces/generic/ScriptInvocationManager.h"
-#include "utils/log.h"
 #include "system.h"
+#include "utils/log.h"
+#include "utils/StringUtils.h"
+
 
 namespace ADDON
 {
 
 std::unique_ptr<CService> CService::FromExtension(CAddonInfo addonInfo, const cp_extension_t* ext)
 {
-  START_OPTION startOption(LOGIN);
+  START_OPTION startOption(START_OPTION::LOGIN);
   std::string start = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@start");
   if (start == "startup")
-    startOption = STARTUP;
-  return std::unique_ptr<CService>(new CService(std::move(addonInfo), TYPE(UNKNOWN), startOption));
+    startOption = START_OPTION::STARTUP;
+  return std::unique_ptr<CService>(new CService(std::move(addonInfo), startOption));
+}
+
+CService::CService(CAddonInfo addonInfo, START_OPTION startOption)
+  : CAddon(std::move(addonInfo)), m_startOption(startOption)
+{
 }
 
+CServiceAddonManager::CServiceAddonManager(CAddonMgr& addonMgr) :
+    m_addonMgr(addonMgr)
+{
+}
 
-CService::CService(CAddonInfo addonInfo, TYPE type, START_OPTION startOption)
-  : CAddon(std::move(addonInfo)), m_type(type), m_startOption(startOption)
+CServiceAddonManager::~CServiceAddonManager()
 {
-  BuildServiceType();
+  m_addonMgr.Events().Unsubscribe(this);
 }
 
-bool CService::Start()
+void CServiceAddonManager::OnEvent(const ADDON::AddonEvent& event)
 {
-  bool ret = true;
-  switch (m_type)
+  if (auto enableEvent = dynamic_cast<const AddonEvents::Enabled*>(&event))
   {
```

Comment:
All these curly braces for a single line block does not match our coding style (throughout the whole `CServiceAddonManager` implementation.